btcjson/v2/btcjson/btcdextcmds.go

```diff
@@ -7,6 +7,42 @@
 
 package btcjson
 
+// NodeSubCmd defines the type used in the addnode JSON-RPC command for the
+// sub command field.
+type NodeSubCmd string
+
+const (
+	// NConnect indicates the specified host that should be connected to.
+	NConnect NodeSubCmd = "connect"
+
+	// NRemove indicates the specified peer that should be removed as a
+	// persistent peer.
+	NRemove NodeSubCmd = "remove"
+
+	// NDisconnect indicates the specified peer should be disonnected.
+	NDisconnect NodeSubCmd = "disconnect"
+)
+
+// NodeCmd defines the dropnode JSON-RPC command.
+type NodeCmd struct {
+	SubCmd        NodeSubCmd `jsonrpcusage:"\"connect|remove|disconnect\""`
+	Addr          string
+	ConnectSubCmd *string `jsonrpcusage:"\"perm|non-perm\""`
```

Comment:
Yeah the intent is to require the caller to specify perm/non-perm. The handler code in `rpcserver.go` currently detects if the optional parameter is missing, and returns the appropriate error if so. 

EDIT: non-perm is now `temp`